lib/decompress/zstd_decompress.c

```diff
@@ -909,6 +909,7 @@ size_t ZSTD_decompressContinue(ZSTD_DCtx* dctx, void* dst, size_t dstCapacity, c
         {   blockProperties_t bp;
             size_t const cBlockSize = ZSTD_getcBlockSize(src, ZSTD_blockHeaderSize, &bp);
             if (ZSTD_isError(cBlockSize)) return cBlockSize;
+            RETURN_ERROR_IF(cBlockSize > ZSTD_BLOCKSIZE_MAX, corruption_detected, "Block Size Exceeds Maximum");
```

Comment:
Yes, this is a first good place for this check.
It controls the compressed size, and no block shall have its compressed size larger than maximum block size.

However : 
- No block shall have its _decompressed_ size larger than maximum block size. This is more complex, and must be controlled elsewhere.
- `ZSTD_BLOCKSIZE_MAX` is the "maximum maximum block size". But the [specifications](https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#blocks) makes it clear that the maximum block size can be smaller than that, depending on frame header instructions. It actually depends on the window size. This value should be computed once, at header decoding stage, stored into `dctx`, and then used for each check.

Reply:
Thanks so much, this was really helpful! I think the maximum block size (min of window size and block size) seemed to already be stored in dctx->fParams.blockSizeMax, so I switched to using that. I've also added a check for the decompressed block size by checking rSize returned after decompressing a block.